hedera-mirror-rest/service/contractService.js

```diff
@@ -209,6 +221,24 @@ class ContractService extends BaseService {
     const rows = await super.getRows(query, params, 'getContractStateChangesByTimestamps');
     return rows.map((row) => new ContractStateChange(row));
   }
+
+  async getContractIdByEvmAddress(evmAddress) {
+    const params = [evmAddress];
+    const rows = await super.getRows(ContractService.contractIdByEvmAddressQuery, params, 'getContractIdByEvmAddress');
+    if (rows.length === 0) {
+      throw new NotFoundError(`No contract with the given evm address: ${evmAddress} has been found.`);
+    }
+    //since evm_address is not an unique index, it is important to make this check.
+    if (rows.length > 1) {
+      throw new Error(`More than one contract with the evm address ${evmAddress} have been found.`);
+    }
+    const contractId = rows[0];
+    return createEntityId(BigInt(contractId.shard), BigInt(contractId.realm), BigInt(contractId.id)).getEncodedId();
```

Comment:
This logic is wrong. The returned id is already an encoded ID that contains shard, realm and num encoded within it. Can adjust the query as well.

```suggestion
    return contractId.id;
```

Reply:
Would even just return `contractId.id` and allow the viewModels to handle formatting

Reply:
Done.